### PR TITLE
Use new lazily initialized value type in projectinfo

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/70991

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Solution/ProjectInfo.cs [hunk 1]
```diff
@@ -496,8 +495,9 @@ internal sealed class ProjectAttributes(
             /// </summary>
             public Guid TelemetryId { get; } = telemetryId;
 
-            private StrongBox<(string?, string?)>? _lazyNameAndFlavor;
-            private Checksum? _lazyChecksum;
+            private SingleInitNullable<(string?, string?)> _lazyNameAndFlavor;
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/Workspace/Solution/ProjectInfo.cs:498]:
```suggestion
            private SingleInitNullable<(string? name, string? flavor)> _lazyNameAndFlavor;
```

**Comment by CyrusNajmabadi** [comment]
@sharwell ptal

**Comment by ToddGrun** [comment]
Are there other usages of StrongBox that we want to change to this too, or is this one unique?

**Comment by CyrusNajmabadi** [comment]
I can look.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**